herddb-core/src/main/java/herddb/sql/CalcitePlanner.java

```diff
@@ -341,6 +341,84 @@ public TranslatedQuery translate(String defaultTableSpace, String query, List<Ob
         }
     }
 
+    private TranslatedQuery calculateShowCreateTable(String query, List<Object> parameters) {
+        String items[] = { "SHOW","CREATE", "TABLE"};
+        if(Arrays.stream(items).allMatch(query::contains)) {
+            query = query.substring(Arrays.stream(items).collect(Collectors.joining(" ")).length()).trim();
+            String tableSpace = "herd";
+            String tableName;
+            if (query.contains(".")) {
+                String tokens[] = query.split("\\.");
+                tableSpace = tokens[0].trim();
+                tableName = tokens[1].trim();
+            } else {
+                tableName = query.trim();
+            }
+            try {
+                Table t = manager.getTableSpaceManager(tableSpace).getTableManager(tableName).getTable();
+                if (t != null) {
+                    StringBuilder sb = new StringBuilder("CREATE TABLE "+tableSpace+"."+tableName);
+                    StringJoiner joiner = new StringJoiner(",","(", ")");
+                    for(Column c : t.getColumns()) {
+                        joiner.add(c.name + " "+ColumnTypes.typeToString(c.type));
+                    }
+                    if(t.getPrimaryKey().length > 0) {
+                        joiner.add("Primary key("+ Arrays.stream(t.getPrimaryKey()).collect(Collectors.joining(",")) + ")");
+                    }
+                    sb.append(joiner.toString());
```

Comment:
You are missing 'create index....' in case of indexes on the table

Reply:
Good catch. I will add those.